src/AsyncWebSynchronization.h

```diff
@@ -7,6 +7,36 @@
 
 #ifdef ESP32
 
+// This is the ESP32 version of the Sync Lock, using the FreeRTOS Semaphore
+// Modified 'AsyncWebLock' to just only use mutex since pxCurrentTCB is not
+// always available. According to example by Arjan Filius, changed name,
+// added unimplemented version for ESP8266
+class AsyncPlainLock
+{
+private:
+  SemaphoreHandle_t _lock;
+  mutable void *_lockedBy;
+
+public:
+  AsyncPlainLock() {
+    _lock = xSemaphoreCreateBinary();
```

Comment:
xSemaphoreCreateBinary() can fail in case of less memory.

Reply:
```suggestion
    _lock = xSemaphoreCreateBinary();
    // If this fails, the system is likely so much out of memory that we should abort
    assert(_lock);
```

Reply:
The original implementation of the AsyncWebLock (below the new class) does the same without an assert etc.

What would you suggest, put an abort() or assert() there? If yes, this should go in both places.

Reply:
assert() is fine and give the developer enough information I think.